client/src/components/SignupForm.js

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from "react";
+import { Input, Button } from "reactstrap";
+import { isEmpty, isLowercase, isEmail, isLength } from "validator";
+
+class SignupForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      fullname: "",
+      username: "",
+      email: "",
+      password: ""
+    };
+  }
+  isRequired = value => !!value;
+  isUserNameValid = value =>
+    isLowercase(value) && isLength(value, { min: 3, max: 12 });
+  isValidPassword = value => isLength(value, { min: 6, max: 16 });
+  handleChange = e => {
+    let { name, value } = e.target;
+    this.setState({ [name]: value });
+  };
+  render() {
+    return (
+      <form onSubmit={this.props.onSubmit}>
+        <div className="form-group">
+          <Input
+            onChange={this.handleChange}
+            value={this.state.fullname}
+            type="text"
+            placeholder="Full Name"
+            name="fullname"
+          />
+          {this.isRequired(this.state.fullname) ? null : (
+            <span className="text-danger">* Is Required</span>
+          )}
+        </div>
+
+        <div className="form-group">
```

Comment:
why not create a generalized component which can have the field and the error message?